packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceProjectCreate.test.ts

```diff
@@ -86,42 +88,47 @@ describe('Force Project Create', () => {
   });
 
   describe('SelectProjectName Gatherer', () => {
-    let inputBoxSpy: sinon.SinonStub;
+    let gatherer: SelectProjectName;
+    const inputBoxStub = sinon.stub(vscode.window, 'showInputBox');
 
-    before(() => {
-      inputBoxSpy = sinon.stub(vscode.window, 'showInputBox');
-      inputBoxSpy.onCall(0).returns(undefined);
-      inputBoxSpy.onCall(1).returns('');
-      inputBoxSpy.onCall(2).returns(PROJECT_NAME);
+    beforeEach(() => {
+      gatherer = new SelectProjectName();
     });
 
     after(() => {
-      inputBoxSpy.restore();
+      inputBoxStub.restore();
```

Comment:
I suggest:
1. move inputBoxStub into each `it()` which requires it
2. remove `inputBoxStub.restore()` from `after()` (and also remove `after()`)

At the moment, here's the lifecycle of inputBoxStub:
* The test starts 
* The instance of the stub is created (line 92)
* test#1 starts (line 102)
* inputBoxStub is set to return the value the test is setting it to (line 103)
* test#1 finishes
(inputBoxStub is not cleaned up)
* test#2 starts
* inputBoxStub is reused
* inputBoxStub is set to return the value the test is setting it to
* test#2 finishes
.
.
.
* test N finishes
* `after()` is called, and `inputBoxStub.restore()` executes


In one of your tests, if you add something like this:

```
    it('Should return cancel if user input is empty string', async () => {
      inputBoxStub.returns('');
      const response = await gatherer.gather();
      expect(response.type).to.equal('CANCEL');
      expect(inputBoxStub.calledOnce).to.be.true; // <<<<<<<<<<<<<<
    });
````

...this is the 3rd test, and I suspect that the "is called once" is going to fail.  For stubs like this, I think it is cleaner to do:

```
describe('SelectProjectName Gatherer', () => {
    let gatherer: SelectProjectName;

    beforeEach(() => {
      gatherer = new SelectProjectName();
    });


    it('Should make one call to showInputBox', async () => {
      const inputBoxStub = sinon.stub(vscode.window, 'showInputBox');
      inputBoxStub.returns(undefined);
      const response = await gatherer.gather();
      expect(inputBoxStub.calledOnce).to.be.true;
    });
.
.
.
```

Reply:
OK - when I try to do this I am getting:

<img width="752" alt="2022-03-25_13-41-32" src="https://user-images.githubusercontent.com/46458081/160198078-5407b610-0c1e-4372-a48e-9db177a9a8ff.png">

Here is the attempted change:
<img width="618" alt="2022-03-25_13-41-20" src="https://user-images.githubusercontent.com/46458081/160198117-24ce5a50-03b7-43da-82f9-7a1cb80835b9.png">

Reply:
Do you know how I can do this and avoid the "already wrapped" error?

Reply:
Thanks for the help, Jeff! I think its right now.